### ISSUE TITLE
Typography

### ISSUE THREAD

**Issue by IanPouncey** [bbc/gel-test-docs#9]
https://bbc.github.io/gel/foundations/typography/

**Comment by IanPouncey** [comment]
## Extrapolation from GEL technical recommendations

### Required

* Focus styles must be visible under all circumstances
* Action dialogs must take focus upon being opened
* Focus can be moved away from any focusable element

### Recommended

* Only interactive elements should be focusable by keyboard
* Only semantic HTML be used for interactive elements
* Avoid layouts in which content/focus order differes from block flow direction
  * Avoid `float` that reverse visual order
  * Avoid `order` property
* Suppress focus styles that appear when an element is clicked for mouse users in browsers that support :focus-visible.
* Remove the default focus style and provide your own
* Provide a reasonably thick outline on focus that matches the text's colour
* Users should be able to press Tab to focus the next interactive element, and Shift and Tab to focus the previous one.
* Avoid positive tabindex values
* Under all circumstances, it should be possible to leave the web page, and the browser, via keyboard

### Implied

* Interactive elements should be focusable by a keyboard
* Interactive elements should be activable using a keyboard

## Suggested ACT expectations

* Focus styles must be visible under all circumstances
* Focus can be moved away from any focusable element
* Only interactive elements should be focusable by keyboard
* Interactive elements must be focusable by a keyboard
* Interactive elements must be activable using a keyboard

**Comment by micmath** [comment]
Re: "Do not increase the letter-spacing of sentence-case text." I think this sounds slightly too general and proscriptive. Maybe add clarity by saying something like "It should be possible for the user to override any typographical settings (size, style, variants etc) if they wish, but the default for sentence text should be Reith and should use the default typographical settings defined in the font." Or summin'?

**Comment by micmath** [comment]
Re "Provide links with an underline to differentiate them" well yes, buhhh.... The BBC has a long history of _not_ doing that, don't they? A quick check just now finds 106 links on the BBC homepage, and nothing appears to have an underline unless you give it hover or focus state. I'm torn on this. The advice is good but it's not GEL in practice. I might change this in the Technical Docs.

**Comment by micmath** [comment]
re "You should not use justified text, instead align text along the left hand side" I would add _for left-to-right text_. Mustn't forget language direction!

**Comment by micmath** [comment]
I've updated the Technical Documentation to clarify. It now says:
> The BBC has traditionally used heavier font-weighting to visually differentiate linked text from surrounding body text, adding an underline that appears on hover or focus. You can follow this established convention by incorporating the underline indicator in your links depending on the hover / focus state, using either `text-decoration: underline` or a border.

**Comment by micmath** [comment]
Considering the comments, this is ready to go.

**Comment by EmmaJP** [comment]
Shouldn't there also be an ACT expectation around font-size being adjustable rather than fixed units?

**Comment by micmath** [comment]
Adjustable in what way? Seriously there are multiple ways that people use to increase legibility by enlarging the effective font size, including zooming (via cmd-plus in a browser). Do you mean that the size is expected to be _set in relative units_? @IanPouncey already lists that under the Recommended heading. Or what specifically is the change you are suggesting?

**Comment by micmath** [comment]
In my experience the most likely problem the user will have when adjusting the font size in a bespoke way (I, for example, have the `prefers-larger-text` option set in my mobile which tends to increase the size of anything textual) is that **the layout** isn't flexible enough to accommodate it. Suggesting that if we document the ability to set larger font-size, the layout must also be self-adjusting to deal with that (within reason).

**Comment by EmmaJP** [comment]
I was just surprised that the recommended relative units isn't one of the ACT expectations. I feel it should be and am curious about why it isn't. 

I'm aware that for many users (like myself) setting a slightly larger default font-size is the only accommodation needed, and it should be respected.

**Comment by micmath** [comment]
@EmmaJP I agree with the recommendation, but was assuming that things listed under "Recommended" would be included in the final deliverable. @IanPouncey can you clarify that I'm understanding that correctly?

I recently read: https://www.24a11y.com/2019/pixels-vs-relative-units-in-css-why-its-still-a-big-deal/
which has some excellent references related to this issue, such as:

* WCAG 1.4.4: Users must be able to resize text without assistive technology up to 200 percent, without loss of content or functionality. (Level AA)
* WCAG 1.4.8: Ideally, we should provide appropriate spacing between lines and paragraphs, and we shouldn’t be requiring the user to scroll horizontally to read a line of text on a full-screen window. (Level AAA)
* WCAG 1.4.10: Users must be able to resize text without being forced to scroll both horizontally and vertically to read that content. (Level AA)

**Comment by micmath** [comment]
I note that while this issue was open, The BBC GEL Design Guidelines have been updated with [information about Spacing Units](https://www.bbc.co.uk/gel/guidelines/spacing-units#what-are-spacing-units-). This excerpt seem relevant:

> Developers at the BBC define space in rem (root em) units. 1 rem is normally equivalent to 16px. Rem units are especially useful for maintaining high accessibility standards as websites coded with rem units will be compatible with browser text resizing. Spacing units are great because they allow designers and developers to speak the same language whilst valuing each discipline's mother tongue, i.e. pixels for a designer and rem units for a developer.

**Comment by EmmaJP** [comment]
Many BBC devs seem to use rem units these days, especially for the more component based stuff that doesn't know where it will be placed in the DOM.

**Comment by micmath** [comment]
@IanPouncey Waiting for changes requested from merge request.
